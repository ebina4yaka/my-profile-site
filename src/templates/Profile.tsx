import { ReactElement } from 'react'
import Copyright from '../atoms/Copyright'
import Navbar from '../organisms/Navbar'
import LanguagesChart from '../organisms/LanguagesChart'
import Hero from '../organisms/Hero'

export default function Profile(): ReactElement {
  return (
    <>
      <header>
        <Navbar />
      </header>
      <main>
        <div className="my-6 grid grid-cols-1 gap-6">
          <Hero />
          <LanguagesChart />
        </div>
      </main>
      <footer className="text-center">
        <Copyright />
      </footer>
    </>
  )
}
