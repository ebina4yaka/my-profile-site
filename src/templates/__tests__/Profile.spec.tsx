import renderer from 'react-test-renderer'
import Profile from '../Profile'

jest.mock('../../organisms/Navbar', () => 'MockNavbar')
jest.mock('../../organisms/Hero', () => 'MockHero')
jest.mock('../../organisms/LanguagesChart', () => 'MockLanguagesChart')
jest.mock('../../atoms/Copyright', () => 'MockCopyright')

it('Snapshot test for Profile', () => {
  const tree = renderer.create(<Profile />).toJSON()
  expect(tree).toMatchSnapshot()
})
