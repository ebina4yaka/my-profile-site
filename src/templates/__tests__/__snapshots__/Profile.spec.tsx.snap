// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Profile 1`] = `
Array [
  <header>
    <MockNavbar />
  </header>,
  <main>
    <div
      className="mt-4"
    >
      <MockHero />
    </div>
    <div
      className="mt-8"
    >
      <MockLanguagesChart />
    </div>
  </main>,
  <footer
    className="text-center"
  >
    <MockCopyright />
  </footer>,
]
`;
